src/main/java/org/apache/kafka/connect/mongodb/MongodbSourceTask.java

```diff
@@ -75,18 +77,25 @@ public void start(Map<String, String> map) {
         topicPrefix = map.get(MongodbSourceConfig.TOPIC_PREFIX);
         uri = map.get(MongodbSourceConfig.URI);
         host = map.get(MongodbSourceConfig.HOST);
-        
+
         try{
             String structConverterClass = map.get(MongodbSourceConfig.CONVERTER_CLASS);
             if(structConverterClass == null || structConverterClass.isEmpty()){
-            	structConverterClass = StringStructConverter.class.getName();
+                structConverterClass = StringStructConverter.class.getName();
             }
             structConverter = (StructConverter) Class.forName(structConverterClass).newInstance();
         }
         catch(Exception e){
-        	throw new ConnectException(MongodbSourceConfig.CONVERTER_CLASS + " config should be a class of type StructConverter");
+            throw new ConnectException(MongodbSourceConfig.CONVERTER_CLASS + " config should be a class of type StructConverter");
+        }
+
+        try{
+            customSchema = Boolean.parseBoolean(map.get(MongodbSourceConfig.CUSTOM_SCHEMA));
```

Comment:
I would suggest making the boolean optional (check for existence with map.containsKey()), and if not specified, default to false to allow the original behavior.  The new behavior will be opt-in then.